src/core/net/srp_client.cpp

```diff
@@ -961,19 +961,16 @@ Error Client::ReadOrGenerateKey(Crypto::Ecdsa::P256::KeyPairAsRef &aKeyRef)
 
     if (error == kErrorNone)
     {
-        Crypto::Ecdsa::P256::PublicKey publicKey;
-
-        if (keyPair.GetPublicKey(publicKey) == kErrorNone)
+        if (aKeyRef.ImportKeyPair(keyPair) != kErrorNone)
         {
-            SuccessOrExit(error = aKeyRef.ImportKeyPair(keyPair));
-            IgnoreError(Get<Settings>().Delete<Settings::SrpEcdsaKey>());
```

Comment:
The new code will not do this `Get<Settings>().Delete<Settings::SrpEcdsaKey>())`? Is this intentional?

I think the idea behind this  is to support transition of device not using key ref (and saving key in `Setting`) to a new build that enabled key-ref feature.

Reply:
It's  missing the `Exitnow()` so will call delete in line 968.
And if the read from Settings failed there is no need to delete the keys.